pkg/operation/botanist/extension.go

```diff
@@ -151,6 +151,10 @@ func (b *Botanist) WaitUntilExtensionResourcesDeleted(ctx context.Context) error
 
 // WaitForExtensionsOperationMigrateToSucceed waits until extension CRs has lastOperation Migrate Succeeded
 func (b *Botanist) WaitForExtensionsOperationMigrateToSucceed(ctx context.Context) error {
+	if err := b.Shoot.Components.Network.WaitMigrate(ctx); err != nil {
+		return err
+	}
+
```

Comment:
There is a lot of duplication between the code that follows and the `WaitUntilExtensionCRMigrated` method - could this method be used below? Or could perhaps the common code be extracted into a helper function to avoid duplication?

Reply:
Yep I should've extracted it in the `common` package similar to the `RestoreExtensionObjectState` method.